File: FISI-TIENDA/Reserva/Reserva/Conexion.py
import psycopg2
import os
from dotenv import load_dotenv
load_dotenv()

class Conexion:
    def __init__(self):
        self.conectar = None

    def establecerConexion(self):
        try:
            self.conectar = psycopg2.connect(
                host = os.environ["HOST"],
                user = os.environ["USER"],
                password = os.environ["PASSWORD"],
                database = os.environ["DATABASE"],
                port = os.environ["PORT"]
            )
            return self.conectar
        except Exception as ex:
            print("Error al conectarse a la base de datos\nError:" + str(ex))

    def ejecutar_update(self,consulta):
        filas_afectadas = 0
        
        print(str(self.conectar))
        
        if self.conectar:
            try:
                cursor = self.conectar.cursor()
                cursor.execute(consulta)
                filas_afectadas = cursor.rowcount
                self.conectar.commit()
                cursor.close()
            except Exception as e:
                print("Error al ejecutar la consulta UPDATE/DELETE/INSERT\nError:", e)
                self.conectar.rollback()
        return filas_afectadas
    
    def ejecutar_insert(self, consulta, valores):
        filas_insertadas = 0
        
        if self.conectar:
            try:
                cursor = self.conectar.cursor()
                cursor.execute(consulta, valores)
                filas_insertadas = cursor.rowcount
                self.conectar.commit()
                cursor.close()
            except Exception as e:
                print("Error al ejecutar la consulta INSERT\nError:", e)
                self.conectar.rollback()
        return filas_insertadas

File: FISI-TIENDA/AdministracionDelInventario/AdministracionDeInventario/src/db/Conexion.java
package db;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class Conexion {
    Connection conectar = null;

    String user = "";
    String password = "";
    String bdname = "";
    String ip = "";
    String port = "";

    String con = "";
    
    private void cargarDatosBD(){
        Properties prop = new Properties();
        InputStream input = null;

        try {
            input = Conexion.class.getClassLoader().getResourceAsStream("resources/config.properties");
            prop.load(input);

            user = prop.getProperty("db.username");
            password = prop.getProperty("db.password");
            bdname = prop.getProperty("db.bdname");
            ip = prop.getProperty("db.ip");
            port = prop.getProperty("db.port");
            
            con = "jdbc:postgresql://" + ip + ":" + port + "/" + bdname;

        } catch (IOException ex) {
            System.out.println("Error en carga de credenciales: "+ex.toString());
        } finally {
            if (input != null) {
                try {
                    input.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }
    
    
    public Connection establecerConexion() {
        cargarDatosBD();
        try {
            Class.forName("org.postgresql.Driver");
            conectar = DriverManager.getConnection(con, user, password);
            System.out.println( "Conexion a la base de datos correcta");
        } catch (Exception e) {
            System.out.println( "Error al conectarse a la base de datos\nError: " + e.toString());
        }
        return conectar;
    }

    public ResultSet ejecutarSelect(String consulta) {
        ResultSet resultado = null;
        try {
            PreparedStatement statement = conectar.prepareStatement(consulta);
            resultado = statement.executeQuery();
        } catch (SQLException e) {
            System.out.println( "Error al ejecutar la consulta SELECT\nError: " + e.toString());
        }
        return resultado;
    }

    public int ejecutarUpdate(String consulta) {
        int filasAfectadas = 0;
        try {
            PreparedStatement statement = conectar.prepareStatement(consulta);
            filasAfectadas = statement.executeUpdate();
        } catch (SQLException e) {
            System.out.println("Error al ejecutar la consulta UPDATE/DELETE/INSERT\nError: " + e.toString());
        }
        return filasAfectadas;
    }
}
